### PR TITLE
feat: :sparkles: change `isFileInFileStorage` rpc return to be more granular

### PR DESCRIPTION
**Pull request by TDemeco**
This PR makes it so instead of returning a boolean (true if the file is found in the File storage, false if it's not), the `isFileInFileStorage` RPC returns three options:
- FileNotFound if the file wasn't found in the File storage.
- IncompleteFile with file metadata, the amount of chunks stored and the total amount of chunks that the file has, if the file is found in the File storage but is still missing chunks.
- Success with the file metadata if the file was found in the File storage and it's complete (not missing any chunks).

### CODE REVIEW

File: client/rpc/src/lib.rs [hunk 1]
```diff
@@ -334,8 +347,24 @@ where
...
         {
-            None => return Ok(false),
-            Some(_) => Ok(true),
+            None => Ok(GetFileFromFileStorageResult::FileNotFound),
+            Some(file_metadata) => {
+                let stored_chunks = read_file_storage
+                    .stored_chunks_count(&file_key)
+                    .map_err(into_rpc_error)?;
+                let total_chunks = file_metadata.chunks_count();
+                if stored_chunks < total_chunks {
```
Comment (snowmead) [inline client/rpc/src/lib.rs:356]: Should we check for an exact match of total chunks as well and return something like `InconsistentFile` (when the number of chunks returned from storage are greater then expected).

**Review by snowmead — APPROVED** [review]
LGTM, just a small concern.

File: client/rpc/src/lib.rs (hunk 1 above)
Comment (links234) [reply client/rpc/src/lib.rs:356]: Good question given the highlighted code, however I'm not sure about codifying the client interface with possible bug scenarios. (Why should I as a client care about you not having a proper file in storage? - just tell me you don't have it and log it for the server operator to see it).

**Review by links234 — APPROVED** [review]
LGTM! Thank you!

File: client/rpc/src/lib.rs [hunk 2]
```diff
@@ -71,6 +71,13 @@ pub enum SaveFileToDisk {
     IncompleteFile(IncompleteFileStatus),
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum GetFileFromFileStorageResult {
```
Comment (ffarall) [inline client/rpc/src/lib.rs:75]: Given that this struct is the exact same as the one above, how about changing the name and reusing it?

**Review by ffarall — APPROVED** [review]

File: client/rpc/src/lib.rs (hunk 1 above)
Comment (snowmead) [reply client/rpc/src/lib.rs:356]: Well as a user, I want to know if my data stored is consistent with what I expect it to be. If it isn't then, I would re-upload the data to ensure it is. We are not just saying if the file exists or not, we are also saying if we are missing data. I think we should also cover the opposite **unlikely** event we have more data then what is expected.

Comment (TDemeco) [reply client/rpc/src/lib.rs:357]: Changed it to account for this!

File: client/rpc/src/lib.rs (hunk 2 above)
Comment (TDemeco) [reply client/rpc/src/lib.rs:75]: With the latest changes it's more differentiated so I think it makes sense to keep both

**Pull request merged**